xds/internal/xdsclient/authority.go

```diff
@@ -424,9 +442,10 @@ func (a *authority) watchResource(rType xdsresource.Type, resourceName string, w
 	if state == nil {
 		a.logger.Debugf("First watch for type %q, resource name %q", rType.TypeEnum(), resourceName)
 		state = &resourceState{
-			watchers: make(map[xdsresource.ResourceWatcher]bool),
-			md:       xdsresource.UpdateMetadata{Status: xdsresource.ServiceStatusRequested},
-			wState:   watchStateStarted,
+			watchers:                make(map[xdsresource.ResourceWatcher]bool),
+			md:                      xdsresource.UpdateMetadata{Status: xdsresource.ServiceStatusRequested},
+			wState:                  watchStateStarted,
+			resourceDeletionIgnored: false,
```

Comment:
Why do we need to set it to `false` explicitly? The zero value of booleans in Go is anyways `false`.